microcosm_pubsub/chain/context.py

```diff
@@ -33,3 +49,41 @@ def __len__(self):
 
     def __getattr__(self, key):
         return self[key]
+
+    def local(self, *args, **kwargs):
+        """
+        Create a locally scoped child context.
+
+        """
+        return ScopedSafeContext(self, *args, **kwargs)
+
+
+class ScopedSafeContext(SafeContext):
```

Comment:
This class implements a scoped context. It defers to the parent when it does not have a value itself.

We do NOT currently allow variable shadowing in the local scope.